File: Clases-Clases-Abstractas-Interfaces/clases-clases-abstractas-interfaces-ethan/src/interfaz/Pato.java
package interfaz;

public class Pato extends Animal implements Volador{

    private String color;

    public Pato(int age, String color) {
        super(age);
        this.color = color;
    }

    @Override
    public void hacerRuido() {
        System.out.println("MUUUU");
    }

    @Override
    public void volar() {
        System.out.println("Volar");
    }
}

File: Clases-Clases-Abstractas-Interfaces/clases-erick/src/animals/Pato.ts
import { Animal } from "./Animal";
import {Volar} from '../skilss/Volar';

export class Pato extends Animal implements Volar {
    constructor(public nombre: string){
        super(nombre);
    }


    hacerSonido():void {
        console.log(`${this.name} cuak!`)
    }

    nadar(): void {
        console.log(`${this.name} nadando`)
    }
}